src/geff/units.py

```diff
@@ -1,10 +1,14 @@
+from __future__ import annotations
+
+from typing import Any, Literal, TypeAlias, TypeGuard, get_args
+
 import numpy as np
+import numpy.typing as npt
 
 # OME-NGFF 0.5 units
 # https://github.com/ome/ngff/blob/7ac3430c74a66e5bcf53e41c429143172d68c0a4/index.bs#L240-L245
 
-VALID_SPACE_UNITS = [
-    None,
+SpaceUnits: TypeAlias = Literal[
```

Comment:
these *could* be used now as pydantic field annotations... but I stopped short of that because it looks like we still want to allow invalid strings and just warn rather than error?